### PR TITLE
UI minor bugfixes & enhancements

### LINKED ISSUE #602: Ignored Resources Show as Failing
### Describe the bug

Suppressed Cloud Security resources show up in the policy overview breakdown as "failing" even though they are marked as "ignored".

### Steps to reproduce

1. Ignore a failing resource
2. Refresh
3. See it shows up under both "Failing" and "Ignored"

### Expected behavior

If the resource is ignored, it should only show up in the "Ignored" filter.

### Environment

- Panther version or commit: v0.3.0-190-g28769b4 (just checked on v1, it's still there)

### Additional context

None

### Screenshots

Failing 1 shows, but 7 resources are listed.

Same thing here, 20 failings, but 22 resources listed.

### LINKED ISSUE #670: Upon first login, minor typo in PW creation helper text
### Describe the bug

When logging in for the first time, you're prompted to set a new password. If the password entered doesn't meet the strength criteria, helper text is displayed to explain the PW policy. 

In the help text, it currently says:

`newPassword must be..`

Should be:

`New password must be..` 

Screenshot below. 

image

### LINKED ISSUE #768: Invalid AlertID still returns data
### Describe the bug

When querying `getAlert` with an invalid AlertID, I expect to get a `null` back or (not preferable) an error. Instead I get the following response back

```
alertId: ""
ruleId: ""
title: ""
creationTime: "0001-01-01T00:00:00Z"
eventsMatched: 0
updateTime: "0001-01-01T00:00:00Z"
eventsLastEvaluatedKey: "eyJsb2dUeXBlVG9Ub2tlbiI6e319"
events: []
```

### Steps to reproduce

1. Invoke `getAlert` supplying an invalid `alertId` parameter
2. Observe the response

### Expected behavior

Return `null` or error out

### Environment

How are you deploying or using Panther?

- Panther version or commit: [e.g. v0.1.1]
- OS: [e.g. MacOS 10.15.3]
- Browser: [e.g. Chrome 80.0.3987.87, Safari Version 13.0.5 (15608.5.11)]

### LINKED ISSUE #812: Change phrasing on unit tests
### Description

In the web UI, the phrasing on the unit tests can for policies & rules can be confusing. The `Expected result is True` and `Expected result is False` radio buttons can be unintuitive to users, and should be update to more directly reflect what is being tested (e.g. `Expected result is True` could be changed to `Test resource is compliant`). Additionally, we could update the green `PASS` and red `FAIL` text could maybe be updated to something like `Test Passed` and `Test Failed`, or something like that.

### Acceptance Criteria

- users are able to intuitively grasp the results of the unit tests via the web UI

### LINKED ISSUE #831: Unable to set the type for additional unit tests
### Describe the bug

I am unable to modify the `Resource Type` or `Log Type` for any unit test beyond the first in the rule/policy editor pages. Instead, when you change the `Resource Type` or `Log Type` for any test besides the first, it changes the first test's type instead.

### Steps to reproduce

1. Go to the policy editor page (either by editing an existing policy or creating a new policy)
2. Scroll down and click the `CREATE TEST` button
3. Attempt to change the `Resource Type`. As soon as you click out of the select box, it reverts back to its default value
4. Click the first unit test
5. Observe that the resource type you had attempted to set on your new test in step 3 has actually been applied to the first unit test

### Expected behavior

I expect that changing the type of a unit test changes that unit tests type, not the first unit tests type

### Environment

How are you deploying or using Panther?

- Web UI, candidate release for [v1.2.0](https://github.com/panther-labs/panther/tree/e3c2761d4157ce9f67bfd6972f78b1f7e9bafcc2)

### Additional context

I thought this was a regression because I could have sworn reporting it before, but when I went looking for the previous instance of this issue I found a similar (but different) one: #210 . I suspect they are related.

### LINKED ISSUE #833: Deleting a Rule returns user to list policies page
### Describe the bug

After deleting a rule from the rule edit page, the user is returned the policy list page.

### Steps to reproduce

1. Go to the rule list page
2. Select a rule
3. Click `DELETE` and `CONFIRM`
4. Observe, you have been magically transported to the policy list page

### Expected behavior

I expect to stay in log analysis land when working on log analysis things.

### Environment

v1.2.0 candidate release

### Additional context

The reverse does not cause any issue, deleting a policy in the same way returns you to the policy list page as expected.

Additionally, I was able to replicate this many times but when going to take the screenshot to show the mismatch the bug actually appeared not to have happened once? Not sure, maybe I hit the back arrow or something.

### Screenshots

Expectations: subverted

Screen Shot 2020-05-05 at 6 58 47 PM

### PR DESCRIPTION
**Pull request by 3nvi**
## Background

This PR solves a lot of minor bugs and adds minor enhancements to the way some edge cases are displayed.

## Changes

- Fix error label on new password. Closes #670 
- Add verbose text to policy/rule tests. Closes #812 
- Ignore suppressed resources/policies in Cloud Security when showing passing/failing. Closes #602 
- Handle non-existent alert, by showing 404 page. Closes #768 
- Fixes issue with invalid redirect. Closes #833 
- Remove all memoization hacks from form components. Closes #831 

## Testing

- Locally

### CODE REVIEW

File: web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx [hunk 1]
```diff
@@ -226,9 +232,9 @@ const BaseRuleFormTestFields: React.FC = () => {
                             value={true}
                           />
                         </Flex>
-                        <Flex justify="space-between" width={225}>
+                        <Flex justify="space-between">
                           <InputElementLabel htmlFor="expected-result-false">
-                            Evaluate to False
+                            Test resource should be non-compliant
```
Comment (kostaspap) [inline web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx:237]: The message should be different for a rule since rules don't have the concept of "resource". My idea is to have ` "Event should trigger alert"/"Event should not trigger alert"`.

Comment (3nvi) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx:237]: 100% right

Comment (3nvi) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx]: Fixed

File: web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx [hunk 2]
```diff
@@ -185,39 +186,46 @@ const BaseRuleFormTestFields: React.FC = () => {
...
+                      columns={3}
+                      gap={10}
                       py={3}
                       borderTop="1px solid"
                       borderBottom="1px solid"
                       borderColor="grey50"
                     >
                       <Flex mt={5}>
-                        <InputElementLabel htmlFor="severity" mr={6}>
+                        <InputElementLabel htmlFor="severity" mr={6} whiteSpace="nowrap">
```
[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx]: 💯

[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestFields/BaseRuleFormTestFields.tsx]: fixed :)

File: web/src/components/forms/BaseRuleForm/BaseRuleFormTestResultList/BaseRuleFormTestResultList.tsx [hunk 3]
```diff
@@ -39,17 +39,29 @@ const BaseRuleFormTestResultList: React.FC<PolicyFormTestResultsProps> = ({ runn
...
+              />
             </Box>
           ))}
           {results.testsErrored.map(({ name: testName, errorMessage }) => (
             <Box key={testName} mb={1}>
-              <PolicyFormTestResult testName={testName} status={ComplianceStatusEnum.Error} />
+              <PolicyFormTestResult
+                testName={testName}
+                status={ComplianceStatusEnum.Error}
+                text="Test Errored"
```
[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestResultList/BaseRuleFormTestResultList.tsx:63]: This is added as a badge, so text needs to be as small as possible. 

I had "PASS", "FAIL", "ERROR" initially which was perfect text for a badge. I'm now forced to change it to something more verbose, but I'm still trying to limit the text.

Any other ideas?

[the remark this answers is not in the document]
Comment (alexmylonas) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestResultList/BaseRuleFormTestResultList.tsx:63]: `Error on Test`?

[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/components/forms/BaseRuleForm/BaseRuleFormTestResultList/BaseRuleFormTestResultList.tsx:63]: Now that I'm re-thinking it, I think that just "ERROR" would do. @nhakmiller wanted the "PASS/FAIL" changed since they are ambiguous, but I feel that "ERROR" is an error no matter what.

Do you think "Error on Test" is even better or are you indifferent?
